# Introduction

Research Question: Which country in South America has the highest and lowest alcohol consumption throughout the years?


In [1]:
import pandas as pd
CD = pd.read_csv('https://raw.githubusercontent.com/NHameleers/dtz2025-datasets/master/CountryHealthFactors.csv')



# Data Preparation

We chose the columns; Alcohol, Country, Year, Population and Status for our research question. The last two were chosen to give an overview of the statistic. We chose all the South American from the database, but French Guyana is missing.

In [2]:
CD_SA = CD.loc[CD['Country'].isin(['Argentina',
'Colombia',
'Venezuela (Bolivarian Republic of)',
'Bolivia (Plurinational State of)',
'Guyana',
'Brazil',
'Chile',
'Suriname',
'Peru',
'Uruguay',
'Paraguay',
'Ecuador']), ['Alcohol', 'Country', 'Year', 'Population', 'Status']]

CD_SA

Alcohol                             Country  Year  Population  \
80        NaN                           Argentina  2015  43417765.0   
81       7.93                           Argentina  2014  42981515.0   
82       8.28                           Argentina  2013  42539925.0   
83       8.35                           Argentina  2012   4296739.0   
84       8.11                           Argentina  2011  41656879.0   
...       ...                                 ...   ...         ...   
2869     7.55  Venezuela (Bolivarian Republic of)  2004         NaN   
2870     6.30  Venezuela (Bolivarian Republic of)  2003         NaN   
2871     6.89  Venezuela (Bolivarian Republic of)  2002         NaN   
2872     8.05  Venezuela (Bolivarian Republic of)  2001         NaN   
2873     8.01  Venezuela (Bolivarian Republic of)  2000         NaN   

          Status  
80    Developing  
81    Developing  
82    Developing  
83    Developing  
84    Developing  
...          ...  
2869  Developing  
2870  Developing  
2871  Developing  
2872  Developing  
2873  Developing  

[192 rows x 5 columns]

# Explore and clean the data

There were 192 rows total before cleaning
We used the.isnull().sum() command to see the ammount of missing data and then proceeded to use the .dropna() command to remove the rows with missing data
When removing the missing values the rows became 180
In the 2014 graph peru and paraguay the value is too small to be visualized and
all the missing values are from 2015


In [3]:
CD_SA.isnull().sum()

Alcohol       12
Country        0
Year           0
Population    32
Status         0
dtype: int64

In [4]:
CD_SA = CD_SA.dropna(thresh=2, subset=['Alcohol', 'Population'])

In [5]:
CD_SA.isnull().sum()

Alcohol       0
Country       0
Year          0
Population    0
Status        0
dtype: int64

# Describe and visualize

To describe our data we choose Status and Population in 2014 for each South American country.

In [6]:
CD_SA.loc[CD_SA['Year'] == 2014, ['Country', 'Year', 'Status', 'Population']]


Country  Year      Status  Population
81    Argentina  2014  Developing  42981515.0
353      Brazil  2014  Developing  24213133.0
545       Chile  2014  Developing  17613798.0
577    Colombia  2014  Developing  47791911.0
787     Ecuador  2014  Developing   1593112.0
1107     Guyana  2014  Developing    763393.0
1992   Paraguay  2014  Developing   6552584.0
2008       Peru  2014  Developing   3973354.0
2474   Suriname  2014  Developing    547928.0
2811    Uruguay  2014  Developing   3419546.0

In [7]:
import altair as alt
import ipywidgets as widgets
from ipywidgets import interact

Then we proceeded to make a chart for each year where we visualize the country and the alcohol consumption

In [8]:
chart = alt.Chart(CD_SA.loc[CD_SA.Year == 2001]).mark_bar().encode(
      y= alt.Y('Country:N', sort='x' ),
      x= ('Alcohol:Q'))
chart

alt.Chart(...)

In [9]:
chart = alt.Chart(CD_SA.loc[CD_SA.Year == 2002]).mark_bar().encode(
      y= alt.Y('Country:N', sort='x' ),
      x= ('Alcohol:Q'))
chart

alt.Chart(...)

In [10]:
chart = alt.Chart(CD_SA.loc[CD_SA.Year == 2003]).mark_bar().encode(
      y= alt.Y('Country:N', sort='x' ),
      x= ('Alcohol:Q'))
chart

alt.Chart(...)

In [11]:
chart = alt.Chart(CD_SA.loc[CD_SA.Year == 2004]).mark_bar().encode(
      y= alt.Y('Country:N', sort='x' ),
      x= ('Alcohol:Q'))
chart

alt.Chart(...)

In [12]:
chart = alt.Chart(CD_SA.loc[CD_SA.Year == 2005]).mark_bar().encode(
      y= alt.Y('Country:N', sort='x' ),
      x= ('Alcohol:Q'))
chart

alt.Chart(...)

In [13]:
chart = alt.Chart(CD_SA.loc[CD_SA.Year == 2006]).mark_bar().encode(
      y= alt.Y('Country:N', sort='x' ),
      x= ('Alcohol:Q'))
chart

alt.Chart(...)

In [14]:
chart = alt.Chart(CD_SA.loc[CD_SA.Year == 2007]).mark_bar().encode(
      y= alt.Y('Country:N', sort='x' ),
      x= ('Alcohol:Q'))
chart

alt.Chart(...)

In [15]:
chart = alt.Chart(CD_SA.loc[CD_SA.Year == 2008]).mark_bar().encode(
      y= alt.Y('Country:N', sort='x' ),
      x= ('Alcohol:Q'))
chart

alt.Chart(...)

In [16]:
chart = alt.Chart(CD_SA.loc[CD_SA.Year == 2009]).mark_bar().encode(
      y= alt.Y('Country:N', sort='x' ),
      x= ('Alcohol:Q'))
chart

alt.Chart(...)

In [17]:
chart = alt.Chart(CD_SA.loc[CD_SA.Year == 2010]).mark_bar().encode(
      y= alt.Y('Country:N', sort='x' ),
      x= ('Alcohol:Q'))
chart

alt.Chart(...)

In [18]:
chart = alt.Chart(CD_SA.loc[CD_SA.Year == 2011]).mark_bar().encode(
      y= alt.Y('Country:N', sort='x' ),
      x= ('Alcohol:Q'))
chart

alt.Chart(...)

In [19]:
chart = alt.Chart(CD_SA.loc[CD_SA.Year == 2012]).mark_bar().encode(
      y= alt.Y('Country:N', sort='x' ),
      x= ('Alcohol:Q'))
chart

alt.Chart(...)

In [20]:
chart = alt.Chart(CD_SA.loc[CD_SA.Year == 2013]).mark_bar().encode(
      y= alt.Y('Country:N', sort='x' ),
      x= ('Alcohol:Q'))
chart

alt.Chart(...)

In the chart for 2014 the values of Peru and Paraguay are too small to be visualized

In [21]:
chart = alt.Chart(CD_SA.loc[CD_SA.Year == 2014]).mark_bar().encode(
      y= alt.Y('Country:N', sort='x' ),
      x= ('Alcohol:Q'))
chart

alt.Chart(...)

In 2015 there were no values for any country as mentioned previously

In [22]:
chart = alt.Chart(CD_SA.loc[CD_SA.Year == 2015]).mark_bar().encode(
      y= alt.Y('Country:N', sort='x' ),
      x= ('Alcohol:Q'))
chart

alt.Chart(...)

We first defined a function called select_year() this was aimed to get a chart of the alcohol consumption of the countries in a specific year.

Furthermore we defined a second function called visualize_year_data() where the chart can be visualized with the chosen year by returning the select_year function.

In [23]:
def select_year(Year):

  return CD_SA.loc[CD_SA['Year'] == Year]


In [24]:
def visualize_year_data(Year):
  data = select_year(Year)

  chart = alt.Chart(data).mark_bar().encode(
      y= alt.Y('Country:N', sort='x' ),
      x= 'Alcohol:Q')

  display(chart)

And to check if this worked we recalled the function visualize_year_data().

In [25]:
visualize_year_data(2009)

alt.Chart(...)

Lastly we used the interact function that was previously imported from widgets to create the interactive chart, where the year can be chosen.

In [26]:
interact(visualize_year_data, Year=[2000, 2001, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010, 2011, 2012, 2013, 2014, 2015])

interactive(children=(Dropdown(description='Year', options=(2000, 2001, 2002, 2003, 2004, 2005, 2006, 2007, 20…

<function __main__.visualize_year_data(Year)>

# Conclusion
We can conclude that among the South American countries, Argentina had the highest alcohol consumption and Ecuador had the lowest alcohol consumption throughout the years!